tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+            'cutmix': 'Only for images',
+            'cutout': 'Only for images',
+            'squeeze_excite': 'Not defined for language model transformers',
+            'mixup': 'Only defined for images',
+            'randaugment': 'Requires PIL dataset to test.',
+            'augmix': 'Required PIL dataset to test.',
+            'progressive_resizing': 'Only defined for images',
+            'stochastic_depth': 'Only applies to ResNets.',
+            'no_op_model': 'Not compatible with this model.'
+        }
+
+
+@pytest.mark.timeout(180)
+@pytest.mark.parametrize("world_size", [
+    pytest.param(1),
+    pytest.param(2, marks=pytest.mark.world_size(2)),
+])
+@pytest.mark.parametrize("device_hparams,deepspeed_enabled,zero_stage", [
+    pytest.param(CPUDeviceHparams(), False, None, id="cpu-ddp"),
+    pytest.param(GPUDeviceHparams(), False, None, id="gpu-ddp", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 0, id="deepspeed-zero0", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 1, id="deepspeed-zero1", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 2, id="deepspeed-zero2", marks=pytest.mark.gpu),
+])
+@pytest.mark.parametrize(
+    "seed,save_interval,save_name_format,resume_file,final_checkpoint",
+    [
+        [None, "1ep", "ep{epoch}", "ep3", "latest-rank{rank}"],  # test randomized seed saving and symlinking
+        [42, "1ep", "ep{epoch}", "ep3", "ep5"],  # test save at epoch end
+        [42, "1ep", "ep{epoch}.tgz", "ep3.tgz", "ep5.tgz"],  # test tarball with compression
+        [42, "2ba", "ba{batch}", "ba12", "ba25"],  # test save batch in partial epoch
+        [42, "1ba", "ba{batch}", "ba15", "ba25"],  # test save final batch in epoch
+        [42, "2ba", "ba{batch}", "ba16", "ba25"],  # test save batch after complete epoch
+    ],
+)
+@pytest.mark.parametrize("model_name", [None, "resnet50_synthetic", "gpt2_52m"])
+@pytest.mark.parametrize("algorithm", algorithms_registry.items(), ids=algorithms_registry.keys())
+def test_algorithm_resumption(
+    device_hparams: DeviceHparams,
+    world_size: int,
+    deepspeed_enabled: bool,
+    zero_stage: Optional[int],
+    composer_trainer_hparams: TrainerHparams,
+    save_interval: str,
+    save_name_format: str,
+    resume_file: str,
+    final_checkpoint: str,
+    seed: Optional[int],
+    model_name: Optional[str],
+    algorithm: Tuple[str, Type[AlgorithmHparams]],
+):
+    """strategy:
+    - train five epochs. capture checkpoints after `checkpoint_interval` and ep5.
+    - create a new trainer from the `checkpoint_interval` checkpoint, and train until end. checkpoint again.
+    - assert that the checkpoint from the new trainer at the end is the same as the checkpoint from the first trainer at the end.
+    """
+    del world_size  # unused. Read via env variable
+
+    algo_name, algo_hparams = algorithm
+
+    if not isinstance(device_hparams, GPUDeviceHparams) and deepspeed_enabled:
```

Comment:
Is this check needed, given that the mark.parameterize above only does deepspeed in combination with GPU?